src/views/open-api-docs-view/components/open-api-overview/index.tsx

```diff
@@ -42,31 +43,37 @@ export function OpenApiOverview({
 	serviceProductSlug,
 	statusIndicatorConfig,
 	contentSlot,
+	versionSwitcherSlot,
 }: OpenApiOverviewProps) {
 	return (
 		<div className={s.overviewWrapper}>
-			<header className={s.header}>
-				<IconTile size="medium" className={s.icon}>
-					<ProductIcon productSlug={serviceProductSlug} />
-				</IconTile>
-				<span>
-					<h1 id={heading.id} className={s.heading}>
-						{heading.text}
-					</h1>
-					{statusIndicatorConfig ? (
-						<Status
-							endpointUrl={statusIndicatorConfig.endpointUrl}
-							pageUrl={statusIndicatorConfig.pageUrl}
-						/>
-					) : null}
-				</span>
-				<Badge
-					className={s.releaseStageBadge}
-					text={badgeText}
-					type="outlined"
-					size="small"
-				/>
-			</header>
+			<div className={s.headerAndVersionSwitcher}>
+				<header className={s.header}>
+					<IconTile size="medium" className={s.icon}>
+						<ProductIcon productSlug={serviceProductSlug} />
+					</IconTile>
+					<span>
+						<h1 id={heading.id} className={s.heading}>
+							{heading.text}
+						</h1>
+						{statusIndicatorConfig ? (
+							<Status
+								endpointUrl={statusIndicatorConfig.endpointUrl}
+								pageUrl={statusIndicatorConfig.pageUrl}
+							/>
+						) : null}
+					</span>
+					<Badge
+						className={s.releaseStageBadge}
+						text={badgeText}
+						type="outlined"
+						size="small"
+					/>
+				</header>
+				{versionSwitcherSlot ? (
+					<div className={s.versionSwitcherSlot}>{versionSwitcherSlot}</div>
```

Comment:
The `OpenApiOverview` component now supports an optional `versionSwitcherSlot`. The work in this component is refactoring the layout a bit to support this change. I've used a `@container` query to try to keep the styling easy to understand across viewports.

Reply:
Slots for the win 🦾